src/main/java/ch/njol/skript/expressions/ExprJoinSplit.java

```diff
@@ -42,16 +42,23 @@
 @Name("Join & Split")
 @Description("Joins several texts with a common delimiter (e.g. \", \"), or splits a text into multiple texts at a given delimiter.")
 @Examples({"message \"Online players: %join all players with \"\" | \"\"%\" # %all players% would use the default \"x, y, and z\"",
-		"set {_s::*} to the string argument split at \",\""})
-@Since("2.1")
+	"set {_s::*} to the string argument split at \",\""})
+@Since("2.1, INSERT VERSION (added regex support)")
 public class ExprJoinSplit extends SimpleExpression<String> {
+	
 	static {
 		Skript.registerExpression(ExprJoinSplit.class, String.class, ExpressionType.COMBINED,
-				"(concat[enate]|join) %strings% [(with|using|by) [[the] delimiter] %-string%]",
-				"split %string% (at|using|by) [[the] delimiter] %string%", "%string% split (at|using|by) [[the] delimiter] %string%");
+			"(concat[enate]|join) %strings% [(with|using|by) [[the] delimiter] %-string%]",
+			"split %string% (at|using|by) [[the] delimiter] %string%",
+			"%string% split (at|using|by) [[the] delimiter] %string%",
+			"regex split %string% (at|using|by) [[the] delimiter] %string%",
+			"regex %string% split (at|using|by) [[the] delimiter] %string%");
 	}
 	
+	private boolean regex;
+	
 	private boolean join;

```

Comment:
1) The extra line is unneeded
2) Since regex is assigned after join, it should be below join here
```suggestion
	private boolean join;
	private boolean regex;

```